cmd/bucket-handlers.go

```diff
@@ -460,6 +473,8 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 		dindex := objectsToDelete[deleteList[i]]
 		apiErr := toAPIError(ctx, errs[i])
 		if apiErr.Code == "" || apiErr.Code == "NoSuchKey" || apiErr.Code == "InvalidArgument" {
+			dObjects[i].DeleteMarkerReplicationStatus = deleteList[i].DeleteMarkerReplicationStatus
+			dObjects[i].VersionPurgeStatus = deleteList[i].VersionPurgeStatus
```

Comment:
Are we guaranteed to hit an error for each object in deleteList? If not the array can have nil errors?